### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by GregoryVds**

### CODE REVIEW

File: CMakeModules/FindCheck.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
```
Comment (matttbe) [inline CMakeModules/FindCheck.cmake]: why do you need this file?
Is it not enough to only use `pkg_check_modules(CHECK REQUIRED check)`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -18,10 +19,15 @@ add_library(convert_util SHARED convert_util.c)
...
 target_link_libraries(convert_client PRIVATE syscall_intercept_static convert_util)
 
+# Unit & integration tests
+add_subdirectory(tests)
+
 # Tests for code linting & formatting
 enable_testing()
 
-set(LIBCONVERT_SOURCES convert_client.c convert.h convert_util.c convert_util.h)
+list(APPEND LIBCONVERT_SOURCES convert_client.c convert.h convert_util.c convert_util.h)
```
Comment (matttbe) [inline CMakeLists.txt:28]: (detail: why do you need a `list`? And why `APPEND`?)

File: CMakeLists.txt [hunk 3]
```diff
@@ -18,10 +19,15 @@ add_library(convert_util SHARED convert_util.c)
...
+
+message("${LIBCONVERT_SOURCES}")
```
Comment (matttbe) [inline CMakeLists.txt]: (do you need this? Or prepend with an explanation)

File: tests/check_convert_util.c [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+main(void)
+{
+	int		number_failed;
+	Suite *		s;
+	SRunner *	sr;
+
+	s	= convert_util_suite();
+	sr	= srunner_create(s);
+
+	srunner_run_all(sr, CK_NORMAL);
```
Comment (matttbe) [inline tests/check_convert_util.c]: maybe `CK_VERBOSE`?

File: tests/check_convert_util.c [hunk 5]
```diff
@@ -31,10 +31,29 @@
...
+
+	hdr.version	= CONVERT_VERSION + 1;
+	ret		= convert_parse_header(buff, sizeof(hdr), &tlvs_length);
+	ck_assert_msg(ret == -1, "Should fail: unsupported version");
+
+	hdr.version		= CONVERT_VERSION;
+	hdr.total_length	= 2;
+	ret			= convert_parse_header(buff, sizeof(hdr),
+	                                               &tlvs_length);
+	ck_assert(ret == 0 && tlvs_length == 4);
```
Comment (matttbe) [inline tests/check_convert_util.c]: can you just comment why `tlvs_length == 4` (and use `ck_assert_msg` with an explanation why it should be that)

**Review by matttbe — CHANGES_REQUESTED** [review]
looks good :-)
Just some questions about the new uni tests ;-)

File: CMakeLists.txt (hunk 3 above)
Comment (GregoryVds) [reply CMakeLists.txt]: No I don't need it! This was added for debugging, I can remove it now :-)

File: CMakeLists.txt (hunk 2 above)
Comment (GregoryVds) [reply CMakeLists.txt:28]: It actually already was a list before: set() creates a list if you pass multiple values.
Now I need to use the more-specific list() call to ask for appending. 

The cmake file in ./tests subdir will also add it own files to this list.
I did not want to duplicate the test logic for linting/code-formatting.
Instead, each sub-directory is in charge of appending the C files that should be cppcheck'ed/uncrustifi'ed to this list.

Comment (matttbe) [reply CMakeLists.txt:28]: OK I see. Maybe better to initialise the list here and append it in the subdirs, no?

File: CMakeModules/FindCheck.cmake (hunk 1 above)
Comment (GregoryVds) [reply CMakeModules/FindCheck.cmake]: Indeed, I don't need it, thx!
I later discovered pkg_check_modules(), and forgot to remove this :-)

File: CMakeLists.txt (hunk 2 above)
Comment (GregoryVds) [reply CMakeLists.txt:28]: The problem is that list() does not support PARENT_SCOPE, so I have to use set() anyway from the subdir.
I use list(APPEND) deliberately in the root CMake file such that I don't have to worry about the order of execution.
There is probably a cleaner way to do that...

Comment (matttbe) [reply CMakeLists.txt:28]: I guess you could create a "function" to call cppcheck and uncrustify from the subdir. I don't know if it helps.
But it is fine for me to keep it like that

Comment (matttbe) [reply CMakeLists.txt:28]: (it is just that I don't know how you would manage the case were you have multiple subdirs)

Comment (GregoryVds) [reply CMakeLists.txt:28]: I am actually doing an append wit set() from the subdir:
set(LIBCONVERT_SOURCES ${LIBCONVERT_SOURCES} tests/check_convert_util.c PARENT_SCOPE)

File: tests/check_convert_util.c (hunk 4 above)
Comment (GregoryVds) [reply tests/check_convert_util.c]: Good idea

File: tests/check_convert_util.c (hunk 5 above)
Comment (GregoryVds) [reply tests/check_convert_util.c]: Sure! Will do.

**Comment by GregoryVds** [comment]
Thanks for the useful review Matt. Merging this as I addressed all your comments :-)

**Pull request merged**